src/reader.cpp

```diff
@@ -3624,6 +3714,32 @@ void reader_data_t::handle_readline_command(readline_cmd_t c, readline_loop_stat
             }
             break;
         }
+        case rl::history_pager: {
+            if (history_pager_active) {
+                fill_history_pager(false, history_search_direction_t::backward);
+                break;
+            }
+
+            const editable_line_t *el = &command_line;
+            const wchar_t *begin, *end;
+            const wchar_t *buff = el->text().c_str();
+            parse_util_token_extent(buff, el->position(), &begin, &end, nullptr, nullptr);
+            wcstring search_string{begin, end};
+
+            // Record our cycle_command_line.
+            cycle_command_line = el->text();
+            cycle_cursor_pos = end - buff;
+
+            this->history_pager_active = true;
+            this->history_pager_history_index_start = 0;
+            this->history_pager_history_index_end = 0;
+            // Update the pager data.
+            pager.set_search_field_shown(true);
+            pager.set_prefix(L"");
```

Comment:
```suggestion
            pager.set_prefix(L"");
            pager.set_fully_disclosed();
```

Discloses immediately.